### PR TITLE
Convert in validations part 2

### PR DESCRIPTION
**Pull request by YuvHayun**
## Related Issues
fixes:  [CIAC-7796](https://jira-dc.paloaltonetworks.com/browse/CIAC-7796)

## Description
Converted move IN validations to new format.

### CODE REVIEW

**Review by GuyAfik — COMMENTED** [review]
Nice job!

File: demisto_sdk/commands/validate/validators/IN_validators/IN106_is_valid_rep_command.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    def validate_rep_commands(self, commands: List[Command]) -> List[str]:
+        invalid_commands = []
+        for command in commands:
+            if command.name in BANG_COMMAND_NAMES:
+                flag_found_arg = False
+                for arg in command.args:
+                    if arg.name in BANG_COMMAND_ARGS_MAPPING_DICT.get(
+                        command.name, {}
+                    ).get("default", []):
+                        flag_found_arg = True
```
[the remark this answers is not in the document]
Comment (YuvHayun) [reply demisto_sdk/commands/validate/validators/IN_validators/IN106_is_valid_rep_command.py:46]: Whether the required argument is found.
for example, if !IP is missing IP argument then we'd like to throw an error.

File: demisto_sdk/commands/validate/validators/IN_validators/IN122_is_valid_feed_integration.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+        Returns:
+            Dict[str, Dict]: The invalid params by name: structure.
+        """
+        invalid_params = {}
+        for required_param in FEED_REQUIRED_PARAMS:
+            if current_param := find_param(params, required_param.get("name", "")):  # type: ignore[arg-type]
+                equal_key_values: dict = required_param.get("must_equal", dict())  # type: ignore[assignment]
+                contained_key_values: dict = required_param.get("must_contain", dict())  # type: ignore[assignment]
+                must_be_one_of: dict = required_param.get("must_be_one_of", list())  # type: ignore[assignment]
```
[the remark this answers is not in the document]
Comment (YuvHayun) [reply demisto_sdk/commands/validate/validators/IN_validators/IN122_is_valid_feed_integration.py:89]: yap

File: demisto_sdk/commands/common/tools.py [hunk 3]
```diff
@@ -4358,3 +4358,11 @@ def get_pack_latest_rn_version(pack_path: str) -> str:
         return list_of_versions[-1]
     else:
         return ""
+
+
+def is_str_bool(input_: str):
```
Comment (GuyAfik) [inline demisto_sdk/commands/common/tools.py:4363]:
```suggestion
def is_str_bool(input_: str) -> bool:
```

File: demisto_sdk/commands/validate/tests/IN_validators_test.py [hunk 4]
```diff
@@ -1914,6 +1947,3487 @@ def test_IsContainingMultipleDefaultArgsValidator_is_valid(
...
+    )
+
+
+def test_IsValidParamDisplayValidator_fix():
+    """
+    Given
+        An integration with 4 params:
+        - one param with invalid display because it's starting with lowercase letter and has underscore.
+        - one param with invalid display because it has underscore.
+        - One param with valid display.
```
Comment (GuyAfik) [inline demisto_sdk/commands/validate/tests/IN_validators_test.py:2067]:
```suggestion
        - one param with valid display.
```

File: demisto_sdk/commands/validate/tests/IN_validators_test.py [hunk 5]
```diff
@@ -1914,6 +1947,3487 @@ def test_IsContainingMultipleDefaultArgsValidator_is_valid(
...
+            "Test.Test_3": "This is test 3 output.",
+        },
+    )
+    results = IsValidEndpointCommandValidator().is_valid(content_items)
+    assert len(results) == expected_number_of_failures
+    assert all(
+        [
+            result.message == expected_msg
+            for result, expected_msg in zip(results, expected_msgs)
+        ]
```
Comment (GuyAfik) [inline demisto_sdk/commands/validate/tests/IN_validators_test.py:3265]: same as above

File: demisto_sdk/commands/common/constants.py [hunk 6]
```diff
@@ -1932,18 +1951,22 @@ class IdSetKeys(Enum):
...
-    STRING = 0
-    NUMBER = 1
-    ENCRYPTED = 4
-    BOOLEAN = 8
-    AUTH = 9
+    STRING = 0  # holds a string argument.
+    NUMBER = 1  # holds an int argument.
+    ENCRYPTED = 4  # holds a password argument.
+    BOOLEAN = 8  # holds a boolean argument (checkbox).
+    AUTH = 9  # holds an auth struct (credentials type).
```
Comment (GuyAfik) [inline demisto_sdk/commands/common/constants.py:1958]: good comments

**Review by GuyAfik — APPROVED** [review]
looks good!

File: demisto_sdk/commands/validate/validators/IN_validators/IN106_is_valid_rep_command.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (YuvHayun) [reply demisto_sdk/commands/validate/validators/IN_validators/IN106_is_valid_rep_command.py:44]: But we also validate that if it's found then the command is valid.. I'll add a description for it.

File: demisto_sdk/commands/validate/validators/IN_validators/IN139_is_name_contain_incident_in_core_pack.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+ContentTypes = Integration
+
+
+class IsNameContainIncidentInCorePackValidator(BaseValidator[ContentTypes]):
+    error_code = "IN139"
+    description = "Validate that there's no 'incident' in any of the commands names or arguments names for core packs integrations."
+    error_message = "The following commands contain the word 'incident' in one or more of their fields, please remove:\n{0}"
+    related_field = "name"
+    invalid_commands: ClassVar[dict[str, dict]] = {}
```
Comment (GuyAfik) [inline demisto_sdk/commands/validate/validators/IN_validators/IN139_is_name_contain_incident_in_core_pack.py:20]:
```suggestion
    invalid_commands: ClassVar[Dict[str, Dict]] = {}
```

Comment (GuyAfik) [reply demisto_sdk/commands/validate/validators/IN_validators/IN139_is_name_contain_incident_in_core_pack.py:20]: check if it has primitive types in other places

**Pull request merged**